src/librustc/session/config.rs

```diff
@@ -695,6 +722,17 @@ pub fn default_configuration(sess: &Session) -> ast::CrateConfig {
     if sess.opts.debug_assertions {
         ret.push(attr::mk_word_item(InternedString::new("debug_assertions")));
     }
+
+    sess.opts.cg.sanitize.map(|s| {
+        let name = match s {
+            Sanitize::Address => "sanitize_address",
+            Sanitize::Leak    => "sanitize_leak",
+            Sanitize::Memory  => "sanitize_memory",
+            Sanitize::Thread  => "sanitize_thread"
+        };
+        ret.push(attr::mk_word_item(InternedString::new(name)));
+    });
```

Comment:
Use case for cfg(sanitizer="leak"). When closing down the application
mem::forget is used to avoid costly deallocation of data structure composed of
parts allocated in different parts of program (so simple blacklisting is not
possible). But developer still would like to use leak sanitizer from time to
time to detect other unintended leaks. So when running under sanitizer
mem::forget could be just skipped.

Similar scenario could happen with thread sanitizer. Application is using
highly optimized inline assembly to implement synchronization primitive.
Thread sanitizer instrumentation does not understand inline assembly and
reports false positives. To avoid this programmer can provide alternative
implementation when using thread sanitizer.

Yes, CLI flags looks better. As far as I understand, it still could be possible
for cfg to be true for multiple values at the same time, right? For example if
in a future it would be possible to enable more than one sanitizer.